ospd_openvas/lock.py

```diff
@@ -29,35 +30,41 @@ class LockFile:
     def __init__(self, path: Path):
         self._lock_file_path = path
         self._has_lock = False
-
-    def is_locked(self) -> bool:
-        return self._lock_file_path.exists()
+        self._fd = None
 
     def has_lock(self) -> bool:
         return self._has_lock
 
     def acquire_lock(self) -> "LockFile":
         """ Acquite a lock by creating a lock file.
         """
-        if self.has_lock() or self.is_locked():
+        if self.has_lock():
             return self
 
         try:
             # create parent directories recursively
             parent_dir = self._lock_file_path.parent
             parent_dir.mkdir(parents=True, exist_ok=True)
 
-            self._lock_file_path.touch(exist_ok=False)
-            self._has_lock = True
-
-            logger.debug("Created lock file %s.", str(self._lock_file_path))
-
-        except FileExistsError as e:
+            self._fd = self._lock_file_path.open('w')
```

Comment:
I think we need to handle other exepction here to and always try to close the fd. also the fd should be set to none in case of errors.